starter-code/src/viking.js

```diff
@@ -1,11 +1,92 @@
 // Soldier
-function Soldier() {}
+class Soldier {
+    constructor(health, strength) {
+        //console.log("Your soldier health is " + health + " and his strength is " + strength);
+        this.health = health;
+        this.strength = strength;
+       // this.damage = health --;
+    }
+
+    attack() {
+        return this.strength
+        console.log("Your attack equals " + this.attack) 
+    }
+
+    receiveDamage(damage) {
+        this.health = this.health - damage
+        // console.log("You've received " + this.damage + " damage")
+    }
+}
+
+var soldierA = new Soldier(300, 150);
+
 
 // Viking
-function Viking() {}
+class Viking extends Soldier {
+  constructor(name, health, strength) {
+    super(health, strength);
+    this.name = name;
+  }
+
+  attack() {
+    return this.strength
+    console.log("Your attack equals " + this.attack)
+  }
+    
+  battleCry() {
+      return "Odin Owns You All!"
+    }
+
+  receiveDamage(damage) {
+    this.health = this.health - damage;
+      if (this.health > 0) {
+          return this.name + " has received " + damage + " points of damage";
+
+      }
+      else {
+          return this.name + " has died in act of combat";
+      }
+  }
+}
+
+
+
 
 // Saxon
-function Saxon() {}
+class Saxon extends Soldier {
+    constructor(health, strength) {
+        super(health, strength);
+    }
+
+    receiveDamage(damage) {
+        this.health = this.health - damage;
+        if (this.health > 0) {
+            return "A Saxon has received " + damage + " points of damage";
+
+        }
+        else {
+            return "A Saxon has died in combat";
+        }
+    }
+
+}
 
 // War
-function War() {}
+class War {
+  constructor() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+  }
+  addViking(Vicking) {
+    this.vikingArmy.push(Vicking);
+  }
+
+  addSaxon(Saxon) {
+    this.saxonArmy.push(Saxon);
+  }
+    vikingAttack() {
+        Saxon.receiveDamage() = vicking.strength();
```

Comment:
For this function to work and pass all the tests you need to:

-get a random Viking from your viking array 
-get a random Saxon from your saxon array 
-When it says "should make a Saxon receiveDamage() equal to the strength of a Viking" this just means that when the viking is attacked the damage points is equal to the strength of the saxon. You need to include something like this  `randomViking.receiveDamage(randomSaxon.strength);`
- you need to slice the array when the saxon dies